src/app/services/local-storage.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { LOCAL_STORAGE_EXPIRATION_TIME } from 'src/api-info';
+
+interface LocalStorageData<T> {
+  timestamp: string;
+  data: T;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalStorageService<T> {
+  public getItem(key: string): T | undefined {
+    const stringValue = localStorage.getItem(key);
+
+    if (stringValue) {
+      const { timestamp, data } = JSON.parse(stringValue);
```

Comment:
I would use try/catch statement for safety

Reply:
will do

Reply:
Updated